apiserver/test/e2e/cluster_server_e2e_test.go

```diff
@@ -0,0 +1,256 @@
+			HttpMethod:       "GET",
+			InputJSON:        "",
+			ExpectedHttpCode: http.StatusOK,
+		},
+		{
+			Name:             "Get a non existing cluster from a namespace",
+			Endpoint:         "/apis/v1alpha2/namespaces/{namespace}/clusters/a-name",
+			HttpMethod:       "GET",
+			InputJSON:        "",
+			ExpectedHttpCode: http.StatusNotFound,
+		},
+		{
+			Name:             "Delete an existing cluster ",
+			Endpoint:         "/apis/v1alpha2/namespaces/{namespace}/clusters/{cluster-name}",
+			HttpMethod:       "DELETE",
+			InputJSON:        "",
+			ExpectedHttpCode: http.StatusOK,
+			ValidationFunc:   waitForDeletedCluster(),
+		},
+		{
+			Name:             "Delete an non existing cluster ",
+			Endpoint:         "/apis/v1alpha2/namespaces/{namespace}/clusters/a-name",
+			HttpMethod:       "DELETE",
+			InputJSON:        "",
+			ExpectedHttpCode: http.StatusNotFound,
+		},
+	}
+
+	// Execute tests sequentially
+	for _, tc := range tests {
+		tc := tc // capture range variable
+		tc.RunTest(t, tCtx)
+	}
+}
+
+func TestClusterServerWithVolumes(t *testing.T) {
+	tCtx, err := NewEnd2EndTestingContext(t)
+	require.NoError(t, err, "No error expected when creating testing context")
+
+	tCtx.CreateComputeTemplate(t)
+	t.Cleanup(func() {
+		tCtx.DeleteComputeTemplate(t)
+	})
+	tests := []End2EndTest{
+		{
+			Name:       "Create cluster with config map volume",
+			Endpoint:   "/apis/v1alpha2/namespaces/{namespace}/clusters",
+			HttpMethod: "POST",
+			InputJSON: `
+			{
+				"name": "{cluster-name}",
+				"namespace": "{namespace}",
+				"user": "boris",
+				"clusterSpec": {
+				  "headGroupSpec": {
+					"computeTemplate": "{compute-template-name}",
+					"image": "{ray-image}",
+					"serviceType": "NodePort",
+					"rayStartParams": {
+					   "dashboard-host": "0.0.0.0",
+					   "metrics-export-port": "8080"
+					 },
+					 "volumes": [
+					   {
+						 "name": "code-sample",
+						 "mountPath": "/home/ray/samples",
+						 "volumeType": "CONFIGMAP",
+						 "source": "{config-map-name}",
+						 "items": {"sample_code.py" : "sample_code.py"}
+					   }
+					 ]
+				  },
+				  "workerGroupSpec": [
+					{
+					  "groupName": "small-wg",
+					  "computeTemplate": "{compute-template-name}",
+					  "image": "{ray-image}",
+					  "replicas": 1,
+					  "minReplicas": 0,
+					  "maxReplicas": 5,
+					  "rayStartParams": {
+						 "node-ip-address": "$MY_POD_IP"
+					   },
+					  "volumes": [
+						{
+						  "name": "code-sample",
+						  "mountPath": "/home/ray/samples",
+						  "volumeType": "CONFIGMAP",
+						  "source": "{config-map-name}",
+						  "items": {"sample_code.py" : "sample_code.py"}
+						}
+					  ]
+					}
+				  ]
+				}
+			}`,
+			ExpectedHttpCode: http.StatusOK,
+			SetupFunc: func(t *testing.T, tCtx *End2EndTestingContext) {
+				// create config map and register a cleanup hook upon success
+				tCtx.CreateConfigMap(t, map[string]string{
+					"sample_code.py": `
```

Comment:
Could `embed.FS` be used to externalise that?

Reply:
Possibly, let me RTFM :)